rustls/src/crypto.rs

```diff
@@ -1,9 +1,14 @@
 use crate::rand::GetRandomFailed;
+use crate::server::ProducesTickets;
 
+use ring::aead;
 use ring::rand::{SecureRandom, SystemRandom};
 
 /// Pluggable crypto galore.
 pub trait CryptoProvider: Send + Sync + 'static {
+    /// Build a ticket generator.
+    fn ticket_generator() -> Result<Box<dyn ProducesTickets>, GetRandomFailed>;
```

Comment:
I think this is an extra indirection that is not needed, because it only serves to customise `rustls::Ticketer` . But that interface just indirects back through `CryptoProvider` having layered on a `TicketSwitcher` which should instead be a building block for concrete ticketers. 

Instead callers of `rustls::Ticketer::new` should just be calling `rustls::crypto::ring::Ticketer::new` or similar.

Reply:
Have added a commit that does this.